src/Controller/CustomNodeEditController.php

```diff
@@ -50,60 +43,50 @@ public static function create(ContainerInterface $container) {
   }
 
   /**
-   * Returns form modes active for current user
-   *
-   * @return array 
+   * Returns form modes active for current user.
    *
+   * @return array
+   *   The list of active modes for the current user.
    */
-  public function getActiveNodeFormModes() {
-    $activeFormModes = [];
-    // "default" form mode is not returned from getFormModes(). Only named form modes.
-    $formModes = $this->entityDisplayRepository->getFormModes('node');
-    // Check which form modes user has access to
-    foreach ($formModes as $key => $value) {
-      if ($this->currentUser()->hasPermission("use {$value['id']} form mode")) {
-        $activeFormModes[$key] = $value;
-      }
-    }
-    
-    return $activeFormModes;
+  public function getActiveNodeFormModes(NodeInterface $node) {
+    $formModes = $this->entityDisplayRepository->getFormModeOptionsByBundle('node', $node->bundle());
+    return array_filter($formModes, function ($mode) {
+      return $this->currentUser()->hasPermission("use node.$mode form mode");
+    });
   }
 
-
   /**
-  * Grants access if user has access to a named form mode, but not default form mode.
-  *
-  * @param \Drupal\Core\Session\AccountInterface $account
-  *   Run access checks for this account.
-  *
-  * @return \Drupal\Core\Access\AccessResultInterface
-  *   The access result.
-  */
-  public function access(AccountInterface $account) {
-    $accessDefaultFormMode = $account->hasPermission('use default form mode');
```

Comment:
So was this also a bug here? `use default form mode` ? instead of  "use $entity_type_id.default form mode"? gosh!!!! Thanks!

Reply:
I was testing it and I must confess I don't fully understand the access method but it does seem to be  $entity_type_id.$mode, according to FormModeManagerBase::setUpUsers